### PR TITLE
Create 2024 weekly template and blocks

### PR DESCRIPTION
**Pull request by jwade1327**
Updated:
screencapture-localhost-22095-templates-pw-weekly-2023-11-14-11_50_25

### CODE REVIEW

**Comment by jwade1327** [comment]
[PW, CMP, HCP, OEM, PFW, Mundo Newsletters & PW Weekly NL 2024](https://trello.com/c/WSn92NAA/649-pw-cmp-hcp-oem-pfw-mundo-newsletters-pw-weekly-nl-2024)

**Review by Shinsina — CHANGES_REQUESTED** [review]
You are going to need to use the `marko-web-query` component using the `all-published-content` query something similar to either of the following: https://github.com/parameter1/pmmi-media-group-websites/pull/588 https://github.com/parameter1/pmmi-media-group-websites/pull/575 you'll likely need to pass the siteId via the query parameters as well as using `includeContentTypes` to include the necessary content types per block.

**Review by Shinsina — CHANGES_REQUESTED** [review]
I'm seeing a lot of new components here with variables being declared within them that go entirely unused, rather than making a comment on each one (since there is a lot of them here). I'm making this blanket comment instead.

Check each file added/updated here and verify that anything that gets declared gets used and moreover verify that the input to each component is only having things passed in that are used, this can be improved by declaring what input a component allows in it's respective `marko.json` file, if you have any questions or need some clarification here let me know :)

File: packages/common/components/blocks/2024-head.marko [hunk 1]
```diff
@@ -0,0 +1,162 @@
+import { get } from "@parameter1/base-cms-object-path";
+
+$ const { website, config } = out.global;
+$ const { newsletter, date } = input;
```
Comment (Shinsina) [inline packages/common/components/blocks/2024-head.marko:4]: `website` and `date` appear to go unused here

File: packages/common/components/blocks/2024/ad/emailx.marko [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { getAsObject, get } from "@parameter1/base-cms-object-path";
```
Comment (Shinsina) [inline packages/common/components/blocks/2024/ad/emailx.marko:1]: `get` appears to go unused here

File: packages/common/components/blocks/2024/ad/emailx.marko [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import defaultValue from "@parameter1/base-cms-marko-core/utils/default-value";
+
+$ const adUnit = getAsObject(input, "adUnit");
+
+$ const useEmailxHref = defaultValue(input.useEmailxHref, false);
+
+$ const { name, alias } = adUnit;
```
Comment (Shinsina) [inline packages/common/components/blocks/2024/ad/emailx.marko:8]: both `name` and `alias` go unused here

File: packages/common/components/blocks/2024/ad/nativex.marko [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { get, getAsObject } from "@parameter1/base-cms-object-path";
+import queryFragment from "../../../../graphql/fragments/content-list";
```
Comment (Shinsina) [inline packages/common/components/blocks/2024/ad/nativex.marko:2]: `queryFragment` goes unused here

File: packages/common/components/blocks/2024/body-wrapper.marko [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+<div class="es-wrapper-color" style="background-color:#F6F6F6">
+  <!--[if gte mso 9]>
+    <v:background xmlns:v="urn:schemas-microsoft-com:vml" fill="t">
+      <v:fill type="tile" color="#f6f6f6"></v:fill>
+    </v:background>
+  <![endif]-->
+  <table class="es-wrapper" width="100%" cellspacing="0" cellpadding="0" style="mso-table-lspace:0pt;mso-table-rspace:0pt;border-collapse:collapse;border-spacing:0px;padding:0;Margin:0;width:100%;height:100%;background-repeat:repeat;background-position:center top;background-color:#F6F6F6">
+    <tr>
+      <td valign="top" style="padding:0;Margin:0">
+        <common-2024-header-block newsletter=newsletter date=date />
```
Comment (Shinsina) [inline packages/common/components/blocks/2024/body-wrapper.marko:17]: I believe date goes unused in `packages/common/components/blocks/2024/header.marko` so date can be removed here

File: packages/common/components/blocks/2024/body-wrapper.marko [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+        <if(input.body)>
+          <${input.body} />
+        </if>
+
+        <common-2024-footer-block newsletter=newsletter date=date />
```
Comment (Shinsina) [inline packages/common/components/blocks/2024/body-wrapper.marko:23]: Both `newsletter` and `date` go unused in `packages/common/components/blocks/2024/footer.marko` so they can likely be removed here

File: packages/common/components/blocks/2024/events.marko [hunk 7]
```diff
@@ -0,0 +1,63 @@
+import queryFragment from "@pmmi-media-group/package-common/graphql/fragments/website-content";
+
+$ const { newsletter, date } = input;
```
Comment (Shinsina) [inline packages/common/components/blocks/2024/events.marko:3]: `newsletter` and `date` go unused here

File: packages/common/components/blocks/2024/header.marko [hunk 8]
```diff
@@ -0,0 +1,39 @@
+import { get } from "@parameter1/base-cms-object-path";
+
+$ const { config } = out.global;
+$ const { newsletter, date } = input;
```
Comment (Shinsina) [inline packages/common/components/blocks/2024/header.marko:4]: `date` goes unused here

File: packages/common/components/blocks/2024/promo.marko [hunk 9]
```diff
@@ -0,0 +1,101 @@
+import { getAsObject } from "@parameter1/base-cms-object-path";
+
+$ const { newsletter, date, content, withImage, sectionName } = input;
```
Comment (Shinsina) [inline packages/common/components/blocks/2024/promo.marko:3]: `newsletter`, `date` and `sectionName` go unused here

File: packages/common/components/blocks/2024/webinar.marko [hunk 10]
```diff
@@ -0,0 +1,95 @@
+import queryFragment from "@pmmi-media-group/package-common/graphql/fragments/website-content";
+
+$ const { newsletter, date } = input;
```
Comment (Shinsina) [inline packages/common/components/blocks/2024/webinar.marko:3]: `newsletter` and `date` go unused here

File: packages/common/components/layouts/2024-weekly.marko [hunk 11]
```diff
@@ -0,0 +1,155 @@
+import { get, getAsArray } from "@parameter1/base-cms-object-path";
+import { stripHtml } from "@parameter1/base-cms-html";
+import queryFragment from "@pmmi-media-group/package-common/graphql/fragments/content-list";
+import moment from "moment";
```
Comment (Shinsina) [inline packages/common/components/layouts/2024-weekly.marko]: getAsArray, stripHtml, queryFragment and moment all go unused

File: packages/common/components/layouts/2024-weekly.marko [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+
+$ const { website, config } = out.global;
+$ const { newsletter, date, sectionName } = input.data;
```
Comment (Shinsina) [inline packages/common/components/layouts/2024-weekly.marko]: `website` and `sectionName` go unused here

File: tenants/all/templates/pw-weekly.marko [hunk 13]
```diff
@@ -0,0 +1,8 @@
+import { get } from "@parameter1/base-cms-object-path";
+
+$ const { newsletter, date } = data;
+$ const { config } = out.global;
+
+$ const newsletterConfig = config.get(newsletter.alias);
+
+<common-2024-weekly-layout data=data />
```
Comment (Shinsina) [inline tenants/all/templates/pw-weekly.marko:8]: `get`, `date` and `newsletterConfig` go unused here. By proxy `newsletter` can also be removed`

**Review by Shinsina — CHANGES_REQUESTED** [review]
Verify that removing these doesn't inherently break anything here, additionally try and ensure input isn't being unnecessarily passed in when it's not used/isn't necessary.

**Review by Shinsina — APPROVED** [review]

**Pull request merged**